mono/metadata/icall.c

```diff
@@ -2962,7 +2982,8 @@ ves_icall_InternalInvoke (MonoReflectionMethod *method, MonoObject *this_arg, Mo
 		mono_security_core_clr_ensure_reflection_access_method (m);
 
 	if (!(m->flags & METHOD_ATTRIBUTE_STATIC)) {
-		if (!mono_class_vtable_full (mono_object_domain (method), m->klass, FALSE)) {
+		if (!mono_class_vtable_full (mono_object_domain (method), m->klass, &error)) {
+			mono_error_cleanup (&error); /* FIXME does this make sense? */
```

Comment:
Not sure the right thing is going on in this branch.  The next line below this one sets `exc` to `mono_class_get_exception_for_failure (m->klass)` (unchanged from the old code).  So it expects that if mono_class_vtable_full fails that `m->klass->exception_type` will be set.  I don't entirely understand what that will always be the case.